sway/commands/seat/hide_cursor.c

```diff
@@ -7,22 +7,41 @@
 
 struct cmd_results *seat_cmd_hide_cursor(int argc, char **argv) {
 	struct cmd_results *error = NULL;
-	if ((error = checkarg(argc, "hide_cursor", EXPECTED_EQUAL_TO, 1))) {
+	if ((error = checkarg(argc, "hide_cursor", EXPECTED_AT_LEAST, 1))) {
 		return error;
 	}
-	if (!config->handler_context.seat_config) {
+	if ((error = checkarg(argc, "hide_cursor", EXPECTED_AT_MOST, 2))) {
+		return error;
+	}
+	struct seat_config *seat_config = config->handler_context.seat_config;
+	if (!seat_config) {
 		return cmd_results_new(CMD_FAILURE, "No seat defined");
 	}
 
-	char *end;
-	int timeout = strtol(argv[0], &end, 10);
-	if (*end) {
-		return cmd_results_new(CMD_INVALID, "Expected an integer timeout");
-	}
-	if (timeout < 100 && timeout != 0) {
-		timeout = 100;
+	if (argc == 1) {
+		char *end;
+		int timeout = strtol(argv[0], &end, 10);
+		if (*end) {
+			return cmd_results_new(CMD_INVALID, "Expected an integer timeout");
+		}
+		if (timeout < 100 && timeout != 0) {
+			timeout = 100;
+		}
+		seat_config->hide_cursor_timeout = timeout;
+	} else {
+		if (strncmp(argv[0], "keyboard", 9)) {
+			return cmd_results_new(CMD_INVALID,
+				"Expected 'hide_cursor <timeout>|keyboard [enable|disable]'");
+		}
+		if (!strncmp(argv[1], "enable", 7)) {
```

Comment:
`strcmp(argv[1], "enable") == 0`

Reply:
Actually now that I'm looking at the code again, I see that there is a `parse_boolean` function in `util.h` which I think fits here perfectly. I will replace the checks here and for "disable" with that function.